data_structures/valid_parens.py

```diff
@@ -0,0 +1,25 @@
+class Solution:
+    def __init__(self):
+        self.stack = []
+
+    def isValid(self, s: str):
+        if not len(s):
+            return True
+        if len(s) % 2:
+            return False
+        for x in s:
+            if not len(self.stack):
```

Comment:
not quite agree here - we should append only "opening" parentheses here - `["(", "{", "["]`

Reply:
that's only for the very 1st append, otherwise we append [here](https://github.com/lisss/ntbts/pull/6/files#diff-6ed49c10ac6d06c9618c59645de8dfcaR22)
not sure how to make it better.. made a few tries but without any luck 😫

Reply:
yeah, but if we get an open parentheses first - we would append it anyway. If we get any closing parentheses first, when stack is empty - we immediately return `False`, as string is already invalid.